homework/index.js

```diff
@@ -30,18 +33,82 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: error.message, class: 'alert-error' });
+  }
+
+  function app(repos) {
+    const root = document.getElementById('root');
+    const container = createAndAppend('div', root, {
+      class: 'container',
+    });
+    const header = createAndAppend('header', container, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    const select = createAndAppend('select', header, {
+      id: 'list',
     });
+    for (let i = 0; i < repos.length; i++) {
+      createAndAppend('option', select, { text: repos[i].name, value: i });
+    }
+    const repoContainer = createAndAppend('div', container, { class: 'container-repo' });
+    const listInfo = createAndAppend('ul', repoContainer, { class: 'repo-info' });
+    const listContributor = createAndAppend('ul', repoContainer, { class: 'repo-contributor' });
+
+    // creating li elements to display repo info when repo name change (onchange event handler)
+    function displayRepoInfo() {
+      removeChildren(listInfo);
+      const repoName = createAndAppend('li', listInfo);
+      repoName.innerHTML = `Repository: <a target=_blank href= ${repos[select.value].html_url}>${
+        repos[select.value].name
+      }</a>`;
+      createAndAppend('li', listInfo, {
+        text: `Description: ${repos[select.value].description}`,
+      });
+      createAndAppend('li', listInfo, { text: `Forks: ${repos[select.value].forks}` });
+      createAndAppend('li', listInfo, {
+        text: `Updated: ${repos[select.value].updated_at}`,
+      });
+      const contributorsUrl = repos[select.value].contributors_url;
+      fetchJSONWithPromise(contributorsUrl)
+        .then(contributorsInfo => {
+          removeChildren(listContributor);
+          contributorsInfo.forEach(contributor => {
+            const contributorName = createAndAppend('li', listContributor);
+            contributorName.innerHTML += `<li><a target =_blank href= ${
```

Comment:
Oops! an `li` inside another `li`.

Reply:
U are right! thanks for your feedback! it's done 👍 :)